juju/juju
# Swap db-accessor cluster configuration source

This patch completes transition of the `dbaccessor` source of truth for controller topology when reasoning about clustering Dqlite.

Reliance on messages broadcast by the `peergrouper` over pub/sub is removed.

Charm-sourced configuration on disk now informs the clustering behaviour of the worker. For further detail on this behaviour, see the [controller charm logic](https://github.com/juju/juju-controller).

At present, the controller charm does not remove data from the cluster configuration when units depart the peer relation, so we cannot QA test the back-stop behaviour here, though the worker logic is unit-tested and considered complete.

A change to the charm will follow.

## QA steps

- Bootstrap to K8s to check the loop-back preference behaviour.
- Bootstrap to LXD with object store configuration allowing HA establishment.
- Check that `juju enable-ha` results in a healthy cluster.

## Documentation changes

None.

## Links

**Jira card:** [JUJU-5308](https://warthogs.atlassian.net/browse/JUJU-5308)



[JUJU-5308]: https://warthogs.atlassian.net/browse/JUJU-5308?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
